exporters/otlp/recordable_test.cc

```diff
@@ -0,0 +1,73 @@
+#include "exporters/otlp/recordable.h"
+
+#include <gtest/gtest.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace otlp
+{
+TEST(Recordable, SetIds)
+{
+  Recordable rec;
+
+  const uint8_t trace_id_buff[trace::TraceId::kSize]      = {0, 0, 0, 0, 0, 0, 0, 0,
+                                                        0, 0, 0, 0, 0, 0, 0, 1};
+  const uint8_t span_id_buff[trace::SpanId::kSize]        = {0, 0, 0, 0, 0, 0, 0, 2};
+  const uint8_t parent_span_id_buff[trace::SpanId::kSize] = {0, 0, 0, 0, 0, 0, 0, 3};
+
+  const trace::TraceId trace_id(
+      nostd::span<const uint8_t, trace::TraceId::kSize>(trace_id_buff, trace::TraceId::kSize));
+  const trace::SpanId span_id(
+      nostd::span<const uint8_t, trace::SpanId::kSize>(span_id_buff, trace::SpanId::kSize));
+  const trace::SpanId parent_span_id(
+      nostd::span<const uint8_t, trace::SpanId::kSize>(parent_span_id_buff, trace::SpanId::kSize));
```

Comment:
https://github.com/open-telemetry/opentelemetry-cpp/pull/127#discussion_r445713527 -

For constants, it will be now compile time error if and of kSize variables changes, while before with 8 and 16 it would still compile but fail later (or not fail but will work incorrectly depending which way size changes). This is part of the reason why it is better to use constants.

In terms of using intermediate buffers for comparison, I think you can also use trace_id.Id().data() which essentially will give the same buffer, so there is no much value to keep them as a separate variable.

Reply:
Makes sense, thanks! I changed it to remove the intermediate buffer and use trace_id.Id().data()